src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs

```diff
@@ -74,6 +74,29 @@ public async Task WriteNestedAsyncEnumerable_DTO<TElement>(IEnumerable<TElement>
             Assert.Equal(1, asyncEnumerable.TotalDisposedEnumerators);
         }
 
+        [Theory]
+        [MemberData(nameof(GetAsyncEnumerableSources))]
```

Comment:
Should there be a non-IAsyncEnumerable test added that can verify a larger, nullable POCO (as a value type)?

Reply:
Nullable POCOs still work, however they don't flow the state. This becomes more evident in the case of IAE which simply fails serialization.